scalding-jdbc-macros/src/main/scala/com/twitter/scalding/jdbc/macros/impl/JdbcMacroImpl.scala

```diff
@@ -0,0 +1,117 @@
+  // Takes a type and its companion objects apply method
+  // based on the args it takes gives back out a field name to symbol
+  private[this] def getDefaultArgs[T](c: Context)(T: c.WeakTypeTag[T]): Map[String, c.Expr[String]] = {
+    import c.universe._
+    val classSym = T.tpe.typeSymbol
+    val moduleSym = classSym.companionSymbol
+    if (moduleSym == NoSymbol) {
+      c.abort(c.enclosingPosition, s"No companion for case class ${T.tpe} available. Possibly a nested class? These do not work with this macro.")
+    }
+    val apply = moduleSym.typeSignature.declaration(newTermName("apply")).asMethod
+    // can handle only default parameters from the first parameter list
+    // because subsequent parameter lists might depend on previous parameters
+    apply.paramss.head.map(_.asTerm).zipWithIndex.flatMap{
+      case (p, i) =>
+        if (!p.isParamWithDefault) None
+        else {
+          val getterName = newTermName("apply$default$" + (i + 1))
+          Some(p.name.toString -> c.Expr(q"${moduleSym}.$getterName.toString"))
+        }
+    }.toMap
+  }
+
+  def caseClassJDBCPayloadImpl[T](c: Context)(implicit T: c.WeakTypeTag[T]): c.Expr[ColumnDefinitionProvider[T]] = {
+    import c.universe._
+
+    if (!IsCaseClassImpl.isCaseClassType(c)(T.tpe))
+      c.abort(c.enclosingPosition, s"""We cannot enforce ${T.tpe} is a case class, either it is not a case class or this macro call is possibly enclosed in a class.
+        This will mean the macro is operating on a non-resolved type.""")
+
+    val defaultArgs = getDefaultArgs[T](c)(T)
+
+    // Field To JDBCColumn
+    def matchField(oTpe: Type,
+      fieldName: FieldName,
+      defaultValOpt: Option[c.Expr[String]],
+      annotationInfo: List[(Type, Option[Int])],
+      nullable: Boolean): scala.util.Try[c.Expr[ColumnDefinition]] = {
+
+      oTpe match {
+        // String handling
+        case tpe if tpe =:= typeOf[String] => StringTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable)
+        case tpe if tpe =:= typeOf[Short] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "SMALLINT")
+        case tpe if tpe =:= typeOf[Int] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "INT")
+        case tpe if tpe =:= typeOf[Long] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "BIGINT")
+        case tpe if tpe =:= typeOf[Double] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "DOUBLE")
+        case tpe if tpe =:= typeOf[Boolean] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "TINYINT")
+        case tpe if tpe =:= typeOf[scala.math.BigInt] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "BIGINT")
+        case tpe if tpe =:= typeOf[java.util.Date] => DateTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable)
+
+        case tpe if tpe.erasure =:= typeOf[Option[Any]] =>
```

Comment:
It would, but i think similar to the TSV problem with a simple flattening to types we can't really support the nested Option[Option[Int]] right? since we won't know at which level the None is